torch/nn/utils/fusion.py

```diff
@@ -97,6 +97,10 @@ def fuse_linear_bn_eval(linear: LinearT, bn: torch.nn.modules.batchnorm._BatchNo
     assert not (linear.training or bn.training), "Fusion only for eval!"
     fused_linear = copy.deepcopy(linear)
 
+    assert not (
+        linear.out_features != bn.num_features and bn.num_features > 1
+    ), "To fuse, linear.out_features == bn.num_features or bn.num_features == 1"
+
```

Comment:
Could either add a test that shows that this would have failed prior or remove this change ? I'm less familiar here and dont want to disrupt

Reply:
Hi @eellison , I've added `test_bn_not_broadcast_with_linear` to check that folding skips (jit frozen path), and fails (`nn.utils.fusion.fuse_linear_bn_eval`) when the dims of linear and bn are not compatible.